src/improvements/tasks/MultisigTask.sol

```diff
@@ -595,133 +605,138 @@ abstract contract MultisigTask is Test, Script, StateOverrideManager {
-            safeTxHash := mload(add(encodedTxData, mload(encodedTxData)))
-        }
-
-        bytes32 domainSeparator = GnosisSafeHashes.calculateDomainSeparator(block.chainid, childMultisig);
         console.log("Domain Hash:    ", vm.toString(domainSeparator));
-        console.log("Message Hash:   ", vm.toString(safeTxHash));
+        console.log("Message Hash:   ", vm.toString(messageHash));
     }
 
     /// @notice print the tenderly simulation payload with the state overrides
-    function printTenderlySimulationData(Action[] memory actions) internal view {
+    function printTenderlySimulationData(Action[] memory actions, address optionalChildMultisig) internal view {
+        // TODO: Support child nonce as a state override. Right now we always get the latest nonce.
+        // Use the max uint256 to indicate that the child multisig nonce is not provided (zero is a valid nonce).
+        uint256 childMultisigNonce =
+            optionalChildMultisig != address(0) ? _getNonce(optionalChildMultisig) : type(uint256).max;
         Simulation.StateOverride[] memory allStateOverrides =
-            getStateOverrides(parentMultisig, _getNonce(parentMultisig));
+            getStateOverrides(parentMultisig, _getNonce(parentMultisig), optionalChildMultisig, childMultisigNonce);
 
-        bytes memory txData = _execTransationCalldata(
-            parentMultisig, getMulticall3Calldata(actions), Signatures.genPrevalidatedSignature(msg.sender)
-        );
+        if (optionalChildMultisig != address(0)) {
+            bytes memory finalExec = getNestedSimulationMulticall3Calldata(actions, optionalChildMultisig);
+
+            console.log("\nSimulation link:");
+            Simulation.logSimulationLink({
+                _to: MULTICALL3_ADDRESS,
+                _data: finalExec,
+                _from: msg.sender,
+                _overrides: allStateOverrides
+            });
+        } else {
+            bytes memory finalExec = _execTransactionCalldata(
+                parentMultisig,
+                getMulticall3Calldata(actions),
+                Signatures.genPrevalidatedSignature(msg.sender),
+                MULTICALL3_ADDRESS
+            );
 
-        // Log the Tenderly JSON payload
-        console.log("\nSimulation payload:");
-        logTenderlySimulationPayload(txData, allStateOverrides[0].overrides);
-
-        // Log the simulation link
-        console.log("\nSimulation link:");
-        Simulation.logSimulationLink({
-            _to: parentMultisig,
-            _data: txData,
-            _from: msg.sender,
-            _overrides: allStateOverrides
-        });
+            // Log the simulation link
+            console.log("\nSimulation link:");
+            Simulation.logSimulationLink({
+                _to: parentMultisig,
+                _data: finalExec,
+                _from: msg.sender,
+                _overrides: allStateOverrides
+            });
```

Comment:
Can I get a call to log out the payload as well, please?

```
            // Log the Tenderly JSON payload
            console.log("\nSimulation payload:");
            logTenderlySimulationPayload(finalExec, allStateOverrides);
```

Reply:
Same comment as above.